packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -256,15 +292,26 @@ export class Cluster extends Resource implements ICluster {
 
     // 0 disables, otherwise forward to underlying implementation which picks the sane default
     if (!options.taskDrainTime || options.taskDrainTime.toSeconds() !== 0) {
-      new InstanceDrainHook(autoScalingGroup, 'DrainECSHook', {
+      this.addInstanceDrainHook(autoScalingGroup, {
         autoScalingGroup,
-        cluster: this,
         drainTime: options.taskDrainTime,
         topicEncryptionKey: options.topicEncryptionKey,
       });
     }
   }
 
+  /**
+   * Create instance draining lifecycle hook for this autoscaling group
+   * @param scope - the scope of this resource
+   * @param options - the options to create the hook
+   */
+  public addInstanceDrainHook(scope: Construct,  options: InstanceDrainHookOpts) {
+    new InstanceDrainHook(scope,'DrainECSHook', {
+      clusterName: this.clusterName,
+      ...options
+    })
+  }
+
```

Comment:
Rather than passing the `cluster` construct, actually we just need the `clusterName` so we pass the `clusterName` here to avoid dependency on the `cluster` construct.